src/planner/binder/statement/bind_create.cpp

```diff
@@ -211,6 +212,36 @@ BoundStatement Binder::Bind(CreateStatement &stmt) {
 		break;
 	}
 	case CatalogType::TABLE_ENTRY: {
+		// If there is a foreign key constraint, resolve primary key column's index from primary key column's name
+		auto &create_info = (CreateTableInfo &)*stmt.info;
+		auto &catalog = Catalog::GetCatalog(context);
+		for (idx_t i = 0; i < create_info.constraints.size(); i++) {
+			auto &cond = create_info.constraints[i];
+			if (cond->type == ConstraintType::FOREIGN_KEY) {
+				auto &fk = (ForeignKeyConstraint &)*cond;
+				if (fk.info.type == ForeignKeyType::FK_TYPE_FOREIGN_KEY_TABLE) {
+					D_ASSERT(fk.info.pk_keys.empty() && !fk.pk_columns.empty());
+					if (create_info.table == fk.info.table) {
+						fk.info.type = ForeignKeyType::FK_TYPE_SELF_REFERENCE_TABLE;
+					} else {
+						// have to resolve referenced table
+						auto pk_table_entry_ptr =
+						    catalog.GetEntry<TableCatalogEntry>(context, fk.info.schema, fk.info.table);
+						if (!pk_table_entry_ptr) {
+							throw ParserException("Can't find table \"%s\" in foreign key constraint", fk.info.table);
+						}
+						D_ASSERT(!fk.pk_columns.empty() && fk.info.pk_keys.empty());
+						for (auto &keyname : fk.pk_columns) {
+							auto entry = pk_table_entry_ptr->name_map.find(keyname);
+							if (entry == pk_table_entry_ptr->name_map.end()) {
+								throw ParserException("column \"%s\" named in key does not exist", keyname);
```

Comment:
This line does not appear to be covered in the code coverage report. Could you add a test that covers this line?